common/datablocks/src/data_block.rs

```diff
@@ -211,8 +211,7 @@ impl DataBlock {
         let mut serializers = vec![];
         for col_index in 0..columns_size {
             let column = self.column(col_index);
-            let field = self.schema().field(col_index);
-            let data_type = field.data_type();
+            let data_type = column.data_type();
```

Comment:
During I implemented outer join, I found the mysql service will hang, but `do_query` already returns data_block.

The root reason is in data block when column data type is nullable, but schema's field data type isn't nullable, then mysql service will hang, so we can just use the column data type to create `TypeSerializerImpl`

Reply:
The problem is sometime `DataField`'s data type doesn't fit with the column's data type, which may cause it choose the wrong serializer. 

While the mismatching behaviour is expected in some cases, since the data type of a column represents physical type, e.g. for `Date` columns the logical type is `Date` and the physical type is `Int32`.

Fortunately, this patch works well because it will always choose the right serializer when using physical type.

Reply:
> While the mismatching behaviour is expected in some cases, since the data type of a column represents physical type, e.g. for `Date` columns the logical type is `Date` and the physical type is `Int32`.

Yes, outer join also needs the mismatching behavior, in databend, when creating a table, the table fields are not nullable by default. But in outer join, they may need null data values in column.

Reply:
> > While the mismatching behaviour is expected in some cases, since the data type of a column represents physical type, e.g. for `Date` columns the logical type is `Date` and the physical type is `Int32`.
> 
> Yes, outer join also needs the mismatching behavior, in databend, when creating a table, the table fields are not nullable by default, but in outer join, they may need null data values in columns.

It's better to cast the columns to nullable.

Reply:
> It's better to cast the columns to nullable.

I have cast columns to nullable.